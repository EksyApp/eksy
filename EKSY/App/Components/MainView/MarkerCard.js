import React, {Component} from 'react';
import {
	View,
	StyleSheet,
	TouchableWithoutFeedback
} from 'react-native';
import {Card, CardSection, Label, TextArea} from '../Common/index'
import Image from 'react-native-fast-image'

export default class MarkerCard extends Component {

	handlePress() {
		this.props.setMarkerSelected(this.props.marker)
		this.props.setMarkerViewVisible()
		this.props.disableGestures(true)
	}

	renderWithImage() {
		return (
				<Card style={[styles.container, {width: this.props.width}]}>
					<CardSection style={[styles.imageContainer, {width: this.props.width}]}>
						<Image resizeMode={Image.resizeMode.cover} style={styles.image} source={{uri: this.props.marker.images[0].uri}}/>
					</CardSection>
					<View style={styles.infoContainer}>
						<Label style={styles.title}>{this.props.marker.title}</Label>
					</View>
				</Card>
		)

	}

	renderWithoutImage() {
		return (
				<Card style={[styles.container, {width: this.props.width}]}>
					<CardSection style={styles.infoContainer}>
						<Label style={styles.title}>{this.props.marker.title}</Label>
					</CardSection>

					<TextArea>{this.props.marker.text}</TextArea>

				</Card>
		)
	}

	renderCard() {
		if (this.props.marker.images && this.props.marker.images.length > 0) {
			return this.renderWithImage()
		} else {
			return this.renderWithoutImage()
		}
	}

	render() {
		return(
				<TouchableWithoutFeedback onPress={() => {this.handlePress()}}>
					{this.renderCard()}
				</TouchableWithoutFeedback>
		)
	}

}

const styles = StyleSheet.create({
	container: {
		height: 200,
		marginLeft: 0,
	},

	imageContainer: {
		width: "100%",
		height: "70%",
		padding: 0
	},

	image: {
		height: "100%",
		width: "100%"
	},

	infoContainer: {
		width: "100%",
		alignItems: "center",
		justifyContent: "center",
		overflow: "scroll"
	},

	title: {
		marginTop: 10
	}

});
