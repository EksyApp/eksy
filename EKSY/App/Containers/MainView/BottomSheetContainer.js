import React, {Component} from 'react'
import BottomSheetComponent from "../../Components/MainView/BottomSheetComponent";
import {connect} from "react-redux";
import * as ReduxActions from "../../Actions";

class BottomSheetContainer extends Component {
	
	render() {
		return (
			<BottomSheetComponent
				markerList={this.props.markerList}
			  setMarkerSelected={this.props.setMarkerSelected}
			  setMarkerViewVisible={this.props.setMarkerViewVisible}
			  disableGestures={this.props.disableGestures}
			/>
		)
	}
	
}

const mapStateToProps = (state) => {
	return {
		markerList: state.markers.markerList,
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		setMarkerSelected: (marker) => {
			dispatch(ReduxActions.setMarkerSelected(marker))
		},
		setMarkerViewVisible: () => {
			dispatch(ReduxActions.setMarkerViewVisible())
		},
		disableGestures: (value) => {
			dispatch(ReduxActions.disableGestures(value))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(BottomSheetContainer)