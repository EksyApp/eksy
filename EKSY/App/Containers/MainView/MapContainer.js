import React, {Component} from 'react'
import * as ReduxActions from "../../Actions/";
import MapComponent from "../../Components/MainView/MapComponent";
import {connect} from "react-redux";

export class MapContainer extends Component {
	
	render() {
		return (
				<MapComponent
						currentRegion={this.props.currentRegion}
						regionChange={this.props.regionChange}
						markerList={this.props.markerList}
						setMarkerSelected={this.props.setMarkerSelected}
						setMarkerViewVisible={this.props.setMarkerViewVisible}
						disableGestures={this.props.disableGestures}
						currentLocation={this.props.currentLocation}
				/>
		)
	}
	
}

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion,
		currentLocation: state.map.location,
		markerList: state.markers.markerList,
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		regionChange: (region) => {
			dispatch(ReduxActions.updateRegion(region))
		},
		setMarkerSelected: (marker) => {
			dispatch(ReduxActions.setMarkerSelected(marker))
		},
		setMarkerViewVisible: () => {
			dispatch(ReduxActions.setMarkerViewVisible())
		},
		disableGestures: (value) => {
			dispatch(ReduxActions.disableGestures(value))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MapContainer)