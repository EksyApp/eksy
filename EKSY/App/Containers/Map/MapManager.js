import React from 'react'
import * as Actions from '../../Actions'
import Store from '../../Store'
import BackgroundGeolocation from 'react-native-mauron85-background-geolocation'

let instance = null
let idCounter = 0

class MapManager {

	constructor() {
		if (!instance) {
			this._map = null
			this._currentLocationMoveRequested = false
			this._reduxState = null
			this.initStore()
			// this.store.subscribe(() => this._storeListener())
			this.startLocationWatcher()
			instance = this
		}
		return instance
	}

	async initStore() {
		this.store = await Store()
		this.store.subscribe(() => this.storeListener())
	}

	static getNextID() {
		idCounter++
		return idCounter
	}

	storeListener() {
		this._reduxState = this.store.getState()
		this.handleFlyingToCurrentLocation()
	}

	handleFlyingToCurrentLocation() {
		if (this._currentLocationMoveRequested) {
			this.goToCurrentPosition()
		}
	}

	startLocationWatcher() {
		BackgroundGeolocation.configure({
			desiredAccuracy: 10,
      stationaryRadius: 10,
      distanceFilter: 10,
      locationTimeout: 30,
      debug: false,
      startOnBoot: false,
      stopOnTerminate: true,
      locationProvider: BackgroundGeolocation.provider.ANDROID_ACTIVITY_PROVIDER,
      interval: 10000,
      fastestInterval: 5000,
      activitiesInterval: 10000,
      stopOnStillActivity: true
		})

		BackgroundGeolocation.on('location', (location) => {
				this.store.dispatch(Actions.updateLocation(location))
				this.store.dispatch(Actions.locationKnown(true))
				this._map.forceUpdate()
		})

		BackgroundGeolocation.on('stationary', () => {})

		BackgroundGeolocation.on('error', (error) => {
			console.log('Geolocation error: ' + error)
			this.store.dispatch(Actions.locationKnown(false))
		})

		BackgroundGeolocation.start(() => {
			console.log('BackgroundGeolocation started')
		})
		// this.watchID = navigator.geolocation.watchPosition(
		// 		(position) => {
		// 			this.store.dispatch(Actions.updateLocation(position.coords))
		// 			this.store.dispatch(Actions.locationKnown(true))
		// 			this._map.forceUpdate()
		// 		},
		// 		(error) => this.store.dispatch(Actions.locationKnown(false)),
		// 		{enableHighAccuracy: false, timeout: 500, maximumAge: 0, distanceFilter: 3}
		// )
	}

	setMapObject(map) {
		this._map = map
	}

	goToCurrentPosition() {
		if (this._reduxState && this._reduxState.map.location.isKnown) {
			this._currentLocationMoveRequested = false
			this.flyToPosition(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
		} else {
			this._currentLocationMoveRequested = true
		}
	}

	flyToPosition(latitude, longitude) {
		let position = {
			latitude: latitude,
			longitude: longitude
		}
		this._map.animateToCoordinate(position, 1000)
	}

}

export default MapManager
