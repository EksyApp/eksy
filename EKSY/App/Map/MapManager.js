import React from 'react'
import * as Actions from '../Actions'
import configureStore from '../Store'
import Marker from './Marker'

let instance = null
let idCounter = 0

class MapManager {

  constructor () {
    if (!instance) {
      this._markers = new Map()
      this._map = null
      this._currentLocationMoveRequested = false
      this._reduxState = null
      this.initStore()
      // this.store.subscribe(() => this.storeListener())
      this.startLocationWatcher()
      instance = this
    }
    return instance
  }

  async initStore () {
    this.store = await configureStore()
    this.store.subscribe(() => this.storeListener())
  }

  static getNextID () {
    idCounter++
    return idCounter
  }

  storeListener () {
    this._reduxState = this.store.getState()
    this.handleFlyingToCurrentLocation()
  }

  handleFlyingToCurrentLocation () {
    if (this._currentLocationMoveRequested) {
      this.goToCurrentPosition()
    }
  }

  startLocationWatcher () {
    navigator.geolocation.watchPosition(
				(position) => {
  this.store.dispatch(Actions.updateLocation(position.coords))
  this.store.dispatch(Actions.locationKnown(true))
  this._map.forceUpdate()
},
				(error) => this.store.dispatch(Actions.locationKnown(false)),
				{enableHighAccuracy: false, timeout: 500, maximumAge: 500, distanceFilter: 10}
		)
  }

  addMarker (key, marker) {
    let markerComponent = <Marker title={marker.title} images={marker.images} latitude={marker.latitude} longitude={marker.longitude} text={marker.text} key={MapManager.getNextID()} />
    this._markers.set(key, markerComponent)
    this._map.forceUpdate()
  }

  removeMarker (key) {
    this._markers.delete(key)
    this._map.forceUpdate()
  }

  getMarkers () {
    return this._markers
  }

  setMapObject (map) {
    this._map = map
  }

  goToCurrentPosition () {
    if (this._reduxState && this._reduxState.map.location.isKnown) {
      this._currentLocationMoveRequested = false
      this.flyToPosition(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
    } else {
      this._currentLocationMoveRequested = true
    }
  }

  flyToPosition (latitude, longitude) {
    let position = {
      latitude: latitude,
      longitude: longitude
    }
    this._map.animateToCoordinate(position, 1000)
  }

}

export default MapManager
