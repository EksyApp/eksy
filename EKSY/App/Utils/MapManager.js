import React from 'react'
import * as Actions from '../Actions/index'
import Store from '../Store/index'
import Dao from "../Dao/Dao";
import {PermissionsAndroid} from 'react-native'
//import BackgroundGeolocation from 'react-native-mauron85-background-geolocation'

let instance = null

// used to give commands to the Google Map
class MapManager {

	constructor() {
		if (!instance) {
			this._map = null
			this._reduxState = null
			this.dao = new Dao()
			this.initStore()
			instance = this
		}
		return instance
	}

	async initStore() {
		this.store = await Store()
		this.store.subscribe(() => this.storeListener())
		this.startLocationWatcher()
	}

	storeListener() {
		this._reduxState = this.store.getState()
	}

	async startLocationWatcher() {
		this._reduxState = this.store.getState()
		
		if(this._reduxState.map.location.isKnown && this._reduxState.map.location.latitude != null) {
			this.dao.updateLocation(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
		}
		
		try {
			const granted = await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION)
			if (granted === PermissionsAndroid.RESULTS.GRANTED) {
				this.watchID = navigator.geolocation.watchPosition(
						(position) => {
							this.dao.updateLocation(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
							this.store.dispatch(Actions.updateLocation(position.coords))
							this.store.dispatch(Actions.locationKnown(true))
							this._map.forceUpdate()
						},
						(error) => this.store.dispatch(Actions.locationKnown(false)),
						{enableHighAccuracy: false, timeout: 500, maximumAge: 0, distanceFilter: 3}
				)
			} else {
				this.store.dispatch(Actions.locationKnown(false))
			}
		} catch (err) {
			this.store.dispatch(Actions.locationKnown(false))
		}
		
		
	}

	setMapObject(map) {
		this._map = map
	}

	goToCurrentPosition() {
		this.flyToPosition(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
	}

	flyToPosition(latitude, longitude) {
		let position = {
			latitude: latitude,
			longitude: longitude
		}
		this._map.animateToCoordinate(position, 1000)
	}

}

export default MapManager
