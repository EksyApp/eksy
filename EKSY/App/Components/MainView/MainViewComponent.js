import React, {Component} from 'react'
import MenuButton from '../../Components/Common/MenuButton'
import {View} from 'react-native'
import {connect} from 'react-redux'
import MarkerViewContainer from "../../Containers/MainView/MarkerViewContainer";
import BottomSheetContainer from "../../Containers/MainView/BottomSheetContainer";
import MapContainer from "../../Containers/MainView/MapContainer";


class MainViewComponent extends Component {
	
	render() {
		return (
				<View>
					<MapContainer />
					<View style={styles.panelContainer}>
						<BottomSheetContainer />
						<MarkerViewContainer/>
					</View>
					<MenuButton onPress={() => {this.props.menuButtonPress()}}/>
				</View>
		)
	}
}


const styles = StyleSheet.create({
	panelContainer: {
		position: 'absolute',
		top: 0,
		bottom: 0,
		left: 0,
		right: 0,
		flex: 1,
	}
})

export default MainViewComponent