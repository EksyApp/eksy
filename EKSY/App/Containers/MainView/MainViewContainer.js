import React, {Component} from 'react'
import * as ReduxActions from '../../Actions/index'
import {connect} from 'react-redux'
import MainViewComponent from "../../Components/MainView/MainViewComponent";
import PropTypes from 'prop-types'

//Renders the main view and handles it's logic
export class MainViewContainer extends Component {
	
	render() {
		return (
				<MainViewComponent
						menuButtonPress = {this.props.menuButtonPress}
				/>
		)
	}
}

MainViewContainer.propTypes = {
	menuButtonPress: PropTypes.func
}

const mapStateToProps = (state) => {
	return {
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		menuButtonPress: () => {
			dispatch(ReduxActions.drawerOpen())
		},
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MainViewContainer)
