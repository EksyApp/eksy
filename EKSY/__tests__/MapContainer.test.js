import React from 'react';
import { MainViewContainer } from '../App/Containers/MainView/MainViewContainer';
import MapManager from '../App/Utils/MapManager';
import renderer from 'react-test-renderer';

MapManager.prototype.startLocationWatcher = jest.fn()
MapManager.prototype.storeListener = jest.fn()
MapManager.prototype.goToCurrentPosition = jest.fn()
MapManager.prototype.flyToPosition = jest.fn()

jest.mock('../App/Containers/MarkerView', () => {
  return require('react-native').View
})

jest.mock('react-native-fetch-blob', () => {
    return {
      DocumentDir: () => {},
      polyfill: () => {}
    }
})

  jest.mock('react-native-snap-carousel', () => {
    const React = require.requireActual('react');
    const {View} = require('react-native')

    class MockCarousel extends React.Component {

      render() {
        return (<View></View>);
      }
    }
    MockCarousel.style = {}
    return MockCarousel
    })

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  class MockMapViewAnimated extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  MockMapView.Animated = MockMapViewAnimated;
  return MockMapView;
});

describe('MainViewContainer renders correctly', () => {
 it('renders correctly', () => {
   const rendered = renderer.create(
     <MainViewContainer currentRegion={{latitudeDelta:1}} currentLocation={{isKnown:false}} markerList={[]} />
   );
   expect(rendered.toJSON()).toMatchSnapshot();
 });
});
