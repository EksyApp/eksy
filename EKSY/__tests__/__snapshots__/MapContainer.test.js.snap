// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer renders correctly renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <MapView
      followUserLocation={true}
      initialRegion={
        Object {
          "latitudeDelta": 1,
        }
      }
      onRegionChange={[Function]}
      showCompass={false}
      showsBuildings={true}
      showsMyLocationButton={true}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          Object {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 50,
            "zIndex": 1,
          },
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 35,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": "rgb(101,114,122)",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î—’
    </Text>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "flex": 1,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <InteractableView
      animatedValueX={undefined}
      animatedValueY={undefined}
      boundaries={
        Object {
          "top": 300,
        }
      }
      dragToss={0.1}
      initialPosition={
        Object {
          "y": 1259,
        }
      }
      onAnimatedEvent={[Function]}
      reportOnAnimatedEvents={true}
      snapPoints={
        Array [
          Object {
            "y": 1039,
          },
          Object {
            "y": 1259,
          },
        ]
      }
      verticalOnly={true}
    >
      <View
        style={
          Object {
            "height": 1259,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 20,
              "position": "relative",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(112,112,112)",
                "borderRadius": 4,
                "height": 8,
                "position": "absolute",
                "top": 30,
                "width": 40,
                "zIndex": 1,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgb(243,244,229)",
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "height": 1259,
              "paddingTop": 20,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 5,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View />
          </View>
        </View>
      </View>
    </InteractableView>
    <View />
  </View>
</View>
`;
